### PR TITLE
Detect if a system variable scope was explicitly specified or not

### PR DESCRIPTION
**Pull request by fulghum**
Changes the `VarScope` function so that it returns whether a scope was explicitly specified, or if one has been inferred.

This is needed because some tooling (e.g. the official MySQL .NET Connector library) will query system variables (e.g. `SELECT @@max_allowed_packet`) and then will look up the returned value in the result set using the expected column name (`@@max_allowed_packet`). Currently, from the way we parse the system variable, this was always returned with the scope present, but to match MySQL's behavior, the column name needs to match the requested name. 

Related GMS PR: https://github.com/dolthub/go-mysql-server/pull/2266/

### CODE REVIEW

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -6497,113 +6497,116 @@ const (
 	SetScope_User        SetScope = "user"
 )
 
-// VarScopeForColName returns the SetScope of the given ColName, along with a new ColName without the scope information.
-func VarScopeForColName(colName *ColName) (*ColName, SetScope, error) {
+// VarScopeForColName returns the SetScope of the given ColName, along with a new ColName without the scope information,
+// and a boolean that indicates if the scope was explicitly specified, or inferred.
+func VarScopeForColName(colName *ColName) (*ColName, SetScope, bool, error) {
```
Comment (zachmu) [inline go/vt/sqlparser/ast.go:6502]: No tests of this function? 😬

File: go/vt/sqlparser/ast.go [hunk 2]
```diff
@@ -6497,113 +6497,116 @@ const (
...
 
 // VarScope returns the SetScope of the given name, broken into parts. For example, `@@GLOBAL.sys_var` would become
 // `[]string{"@@GLOBAL", "sys_var"}`. Returns the variable name without any scope specifiers, so the aforementioned
-// variable would simply return "sys_var". `[]string{"@@other_var"}` would return "other_var". If the name parts do not
+// variable would simply return "sys_var". `[]string{"@@other_var"}` would return "other_var". If a scope is not
+// explicitly specified, then the boolean return parameter will be false. If the name parts do not
 // specify a variable (returns SetScope_None), then it is recommended to use the original non-broken string, as this
 // will always only return the last part. `[]string{"my_db", "my_tbl", "my_col"}` will return "my_col" with SetScope_None.
-func VarScope(nameParts ...string) (string, SetScope, error) {
+func VarScope(nameParts ...string) (string, SetScope, bool, error) {
```
Comment (zachmu) [inline go/vt/sqlparser/ast.go:6544]: Maybe a little cleaner to add this as a new field on SetScope rather than as a bool here

**Review by zachmu — APPROVED** [review]
LGTM!

File: go/vt/sqlparser/ast.go (hunk 2 above)
Comment (fulghum) [reply go/vt/sqlparser/ast.go:6544]: Good idea. `SetScope` is currently just an alias for a `string`, so we'd need to turn it into a struct. Might not be worth it right this minute, but I'll take a closer look.

File: go/vt/sqlparser/ast.go (hunk 1 above)
Comment (fulghum) [reply go/vt/sqlparser/ast.go:6502]: Good catch! Added unit tests in bd94646f66fce9b11dd3b302e7836a09b4328a42

**Pull request merged**